Assets/Scripts/Minigames/CowardMiniGame/WrenchBehavior.cs

```diff
@@ -0,0 +1,187 @@
+/*****************************************************************************
+// File Name :         WrenchBehavior.cs
+// Author :            Mark Hanson
+// Contributors :      Marissa Moser
+// Creation Date :     5/27/2024
+//
+// Brief Description : Any function to do with the wrench will be found here. Wrench swinging, spark interaction, and completion of this segment of the minigame.
+*****************************************************************************/
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+using UnityEngine.UI;
+using PlaceboEntertainment.UI;
+using System;
+
+public class WrenchBehavior : MonoBehaviour, IInteractable
+{
+    [Header("UI Stuff")]
+    [SerializeField] private TextMeshPro _smackedText;
+
+    [Header("Wrench overall functions")]
+    [SerializeField] private GameObject _sparksMode;
+    //private PlayerController _pc;
+    private int _sparkSmacked;
+
+    [Header("Wrench within hand functions")]
+    [SerializeField] private Animator _animate;
+    [SerializeField] private GameObject _wrenchSpark;
+    //private bool _swing;
+    [Header("Wrench outside hand functions")]
+    [SerializeField] private GameObject _rightHand;
+    private bool _withinProx;
+    private bool _isEquipped;
+
+    public static Action SparkSmackedAction;
+
+    void Awake()
+    {
+        _rightHand = GameObject.FindWithTag("Righty");
+        _sparksMode = GameObject.Find("SparksMode");
+        GameObject _smackTextObject = GameObject.Find("Spark num");
+        _smackedText = _smackTextObject.GetComponent<TextMeshPro>();
+    }
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        //GameObject _pcObject = GameObject.FindWithTag("Player");
+        //_pc = _pcObject.GetComponent<PlayerController>();
+        _isEquipped = false;
+        //_withinProx = false;
+        //_swing = false;
+        SparkSmackedAction += SparkSmacked;
+    }
+    void FixedUpdate()
+    {
+        //if (_interact && _isEquipped == false && _withinProx == true)
+        //{
+        //    transform.parent = _rightHand.transform;
+        //    _isEquipped = true;
+        //}
+        if(_isEquipped == true)
+        {
+            transform.position = new Vector3(_rightHand.transform.position.x, _rightHand.transform.position.y, _rightHand.transform.position.z);
+            transform.rotation = _rightHand.transform.rotation;
+        }
+        //if(_interact && _isEquipped == true)
+        //{
+        //    StartCoroutine(Swinging());
+        //}
+        //Debug.Log(_isEquipped);
+    }
+    // Update is called once per frame
+    void Update()
```

Comment:
This Update function could be deleted to clean up the code a bit if it's not being used.

Reply:
wrench script cleaned up